docs/usage.md

```diff
@@ -249,3 +266,38 @@ initializes and makes use of TLS. If you're seeing compartment violations when
 accessing TLS then you likely need to run `pad-tls` on the relevant. The tool
 can be found at `$IA2_PATH/build/tools/pad-tls/pad-tls`. Note that the `ldd` and
 `lddtree` tools can be used to list the DSO dependencies of your app.
+
+## Hidden Symbols and Linker Errors
+
+When building the post-rewriter compartmentalized version of your project you
+may run into linker errors where `libcallgates.so` can't find symbols within a
+compartment. There are a couple of reasons why that may happen.
+
+As mentioned in the rewriter section, if a function is `static`, but the
+`static` keyword comes from inside a macro, then the rewriter will fail to emit
+an `__attribute__((used))` annotation for the function. This can lead to linker
+errors if the function is used as an address-taken function across compartments.
+
+Similar issues can occur when using linker scripts. If your project uses a
+linker script to mark functions local, you'll run into linker errors where the
```

Comment:
What's meant by "local" here exactly? `static`, or does it mean something similar but slightly different?

Reply:
I don't know the canonical definition of the notion of a local symbol, but this link is a decent starting point: https://sourceware.org/binutils/docs/ld.html#VERSION

At the very least, one difference is that `static` is a C notion, while local symbols are a linker notion.

Reply:
It might be clearer to say "mark function **symbols** as local".